src/obr/cli.py

```diff
@@ -373,25 +379,34 @@ def archive(ctx: click.Context, **kwargs):
     jobs = filter_jobs_by_query(project, filters)
 
     time = str(datetime.now()).replace(" ", "_")
-    use_github_repo = False
     repo = None
     branch_name = None
     previous_branch = None
     # check if given path is actually a github repository
+    use_git_repo = False
     try:
         repo = Repo(path=str(target_folder), search_parent_directories=True)
-        previous_branch = repo.active_branch.name
-        branch_name = "archive-" + (
-            str(datetime.now()).rsplit(":", 1)[0].replace(" ", ":").replace(":", "_")
-        )
-        use_github_repo = True
-        logging.info(f"checkout archive-{branch_name}")
-        repo.git.checkout("HEAD", b=branch_name)
+        use_git_repo = True
     except InvalidGitRepositoryError:
         logging.warn(
             f"Given directory {target_folder=} is not a github repository. Will only"
             " copy files."
         )
+    if use_git_repo:
+        previous_branch = repo.active_branch.name
+        if branch := kwargs.get("amend"):
+            use_github_repo = True
+            logging.info(f"checkout {branch_name}")
+            repo.git.checkout(branch_name)
```

Comment:
For the sake of error handling, it is probably useful to check if the branch actually exists.
```suggestion
            if not branch in repo.git.branch():
                    # throw error, return 
                    logging.error(f"Specified 'amend' branch does not exist. Current existing branches include {repo.git.branch()}.")
                    return
            repo.git.checkout(branch_name)
```

..or, alternatively, wrap this inside another `try-except GitCommandError` block.